src/mame/includes/cgc7900.h

```diff
@@ -81,19 +95,50 @@ class cgc7900_state : public driver_device
 	DECLARE_WRITE16_MEMBER( z_mode_w );
 	DECLARE_WRITE16_MEMBER( color_status_w );
 	DECLARE_READ16_MEMBER( sync_r );
+	DECLARE_READ16_MEMBER( unmapped_r );
+
+	DECLARE_WRITE_LINE_MEMBER(irq_0);
+	DECLARE_WRITE_LINE_MEMBER(irq_1);
+	DECLARE_WRITE_LINE_MEMBER(irq_2);
+	DECLARE_WRITE_LINE_MEMBER(irq_3);
+	DECLARE_WRITE_LINE_MEMBER(irq_4);
+	DECLARE_WRITE_LINE_MEMBER(irq_5);
+	DECLARE_WRITE_LINE_MEMBER(irq_6);
+	DECLARE_WRITE_LINE_MEMBER(irq_7);
+	DECLARE_WRITE_LINE_MEMBER(irq_8);
+	DECLARE_WRITE_LINE_MEMBER(irq_9);
+	DECLARE_WRITE_LINE_MEMBER(irq_a);
+	DECLARE_WRITE_LINE_MEMBER(irq_b);
+	DECLARE_WRITE_LINE_MEMBER(irq_c);
+	DECLARE_WRITE_LINE_MEMBER(irq_d);
+	DECLARE_WRITE_LINE_MEMBER(irq_e);
+	DECLARE_WRITE_LINE_MEMBER(irq_f);
+
+	DECLARE_WRITE8_MEMBER(baud_write);
+	DECLARE_WRITE_LINE_MEMBER(write_rs232_clock);
+	DECLARE_WRITE_LINE_MEMBER(write_rs449_clock);
 
 	void update_clut();
 	void draw_bitmap(screen_device *screen, bitmap_rgb32 &bitmap);
 	void draw_overlay(screen_device *screen, bitmap_rgb32 &bitmap);
 
 	/* interrupt state */
-	uint16_t m_int_mask;
+	uint16_t m_int_mask, m_int_active;
 
 	/* video state */
 	rgb_t m_clut[256];
 	int m_blink;
 
 	TIMER_DEVICE_CALLBACK_MEMBER(blink_tick);
+
+	void kbd_put(u8 data);
+
+private:
+	u16 kbd_mods;
```

Comment:
Is there any reason you've only made this stuff private and not a whole lot of other stuff as well?  There's no reason for all the device members to be public.

Reply:
It's used by MCFG_GENERIC_KEYBOARD_CB macro and has to be public.

Reply:
I was referring to all the `required_device` and `required_shared_ptr` stuff above, and the overrides of protected member functions.  I realise the keyboard put callback needs to be public.  I know it was like that already, but it's nice to do "drive-by cleanup" when doing stuff like this.